### PR TITLE
Documentation update for Redis backed github-mirror

### PR DESCRIPTION
**Pull request by m-ashour**
Signed-off-by: Maha Ashour <mashour@redhat.com>

### CODE REVIEW

File: docs/devel.md [hunk 1]
```diff
@@ -146,6 +146,57 @@ The service log will show:
...
+You will also need to set up the following environment variables:
+```
+REDIS_ENDPOINT = <redis.endpoint>
+REDIS_PORT = <port>
+REDIS_PASSWORD = <mysecret>
+```
+
+where `REDIS_PASSWORD` is the same password set in the Redis configuration file. `REDIS_ENDPOINT` and `REDIS_PORT` are the address and port of the Redis server (typically this is `localhost` and `6379`).
+
+[Please note that github-mirror expects an encrypted SSL connection to Redis, so ensure you are connecting to Redis via an SSL encrypted tunnel or manually disable SSL encryption [here](https://github.com/app-sre/github-mirror/blob/master/ghmirror/data_structures/redis_data_structures.py#L36)].
```
Comment (m-ashour) [inline docs/devel.md:196]: I kind of hate this. Ideally I think that ssl encryption should be an environment varaible whose value is a parameter in the saas file and defaults to True. That way when running this locally, I can simply set the environment variable to false without editing the code.

File: docs/devel.md [hunk 2]
```diff
@@ -146,6 +146,57 @@ The service log will show:
...
+```
+
+Use the Redis client `redis-cli` to test your connection to the Redis server.
+
+By default, `redis-cli` will connect to the server at `127.0.0.1` port `6379`:
+```
+user@localhost:~$ redis-cli -a <mysecret>
+127.0.0.1:6379> ping
+PONG
+```
```
Comment (apahim) [inline docs/devel.md:177]: I'm not sure we should approach redis config here, even if in a very simplistic way... I'd rather just have a docker command line, like:
```
docker run --rm -it -p 6379:6379 redis
```
That's is enough for testing.

File: docs/devel.md [hunk 3]
```diff
@@ -146,6 +146,57 @@ The service log will show:
...
+
+### B. Configuring github-mirror to run in Redis mode
+
+First you will need to tell github-mirror to run in Redis mode by setting the `CACHE_TYPE` environment variable. If the variable is not set, github mirror will run in the default in-memory cache mode.
+
+```
+CACHE_TYPE = redis
```
Comment (apahim) [inline docs/devel.md:184]: $ export CACHE_TYPE=redis

File: docs/devel.md [hunk 4]
```diff
@@ -146,6 +146,57 @@ The service log will show:
...
+```
+
+You will also need to set up the following environment variables:
+```
+REDIS_ENDPOINT = <redis.endpoint>
+REDIS_PORT = <port>
+REDIS_PASSWORD = <mysecret>
```
Comment (apahim) [inline docs/devel.md:191]: same here...
```
$ export REDIS_ENDPOINT=...
...
```

File: docs/devel.md (hunk 1 above)
Comment (apahim) [reply docs/devel.md:196]: Will it work with the simplest redis configuration for testing?

Comment (m-ashour) [reply docs/devel.md:196]: you mean like running make check?

**Comment by m-ashour** [comment]
I'm not sure we can support SSL for testing without having the user set up an SSL encrypted tunnel like this:
https://redislabs.com/blog/stunnel-secure-redis-ssl/#:~:text=The%20stunnel%20application%20is%20a,will%20never%20be%20captured%20unencrypted.

Should I recreate those steps in the documentation?

File: docs/devel.md (hunk 2 above)
Comment (m-ashour) [reply docs/devel.md:177]: I don't think you can set the authentication that way. you'd still have to supply a config file.

and you'd still want to use the redis client to test the connection...

File: docs/devel.md (hunk 1 above)
Comment (m-ashour) [reply docs/devel.md:196]: but yes if I pull the value of ssl from an environment variable, i can test `curl` requests to github mirror

Comment (apahim) [reply docs/devel.md:196]: Yeah, I mean, you propose in the docs to run redis in a given way. But the source code won't work with that redis. Docs should give some steps that actually work. So, if SSL is a requirement, it's ok to say in the docs "go deploy redis with ssl then come back here" :)
Or, as you said, pull the ssl value from the environment so a developer can have a simple env to play with.

File: docs/devel.md (hunk 2 above)
Comment (apahim) [reply docs/devel.md:177]: I see... there's more to it indeed.

**Comment by apahim** [comment]
You can just point to some docs that tell how to do it. My point being you don't have to teach people how to install/configure redis.

File: docs/devel.md [hunk 5]
```diff
@@ -146,6 +146,58 @@ The service log will show:
...
+127.0.0.1:6379> ping
+PONG
+```
+
+### B. Configuring github-mirror to run in Redis mode
+
+First you will need to tell github-mirror to run in Redis mode by setting the `CACHE_TYPE` environment variable. If the variable is not set, github mirror will run in the default in-memory cache mode.
+
+```
+user@localhost:~$ export CACHE_TYPE = redis
```
Comment (apahim) [inline docs/devel.md:184]: no whitespaces around the equal sign

File: docs/devel.md (hunk 1 above)
Comment (m-ashour) [reply docs/devel.md]: great advice as usual, i hope you like the new and improved version

File: docs/devel.md (hunk 5 above)
Comment (m-ashour) [reply docs/devel.md]: ekkk. thanks for catching this.

**Pull request merged**